src/printer.js

```diff
@@ -551,7 +551,12 @@ function genericPrintNoParens(path, options, print, args) {
       // if the arrow function is expanded as last argument, we are adding a
       // level of indentation and need to add a softline to align the closing )
       // with the opening (.
-      const shouldAddSoftLine = args && args.expandLastArg;
+      // or if the arrow functino is inside a JSXExpression (e.g. an attribute)
+      // we should align the expression's closing } with the line with the
+      // opening {.
+      const shouldAddSoftLine =
+        (args && args.expandLastArg) ||
+        path.getParentNode().type === "JSXExpressionContainer";
```

Comment:
This is a much better fix, thanks!

Could you make sure that you tests both arrow functions inside of attribute values and children? They share the same JSXExpressionContainer type.

Reply:
Typo: `functino`